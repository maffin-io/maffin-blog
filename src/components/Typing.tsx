import React from 'react';
import Link from 'next/link';

export default function Typing(): JSX.Element {
  return (
    <p className="overflow-hidden whitespace-nowrap font-mono text-lg leading-7 text-gray-500 animate-typing">
      <span>
        everything
        {' '}
      </span>
      <span className="text-emerald-500">
        <Link href="/tags/travel">travel</Link>
      </span>
      <span>
        ,
        {' '}
      </span>
      <span className="text-sky-500">
        <Link href="/tags/engineering">engineering</Link>
      </span>
      <span>
        {' '}
        and
        {' '}
      </span>
      <span className="text-orange-500">
        <Link href="/tags/finance">finance</Link>
      </span>
    </p>
  );
}
