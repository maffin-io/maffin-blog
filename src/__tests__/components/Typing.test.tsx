import { render } from '@testing-library/react';

import Typing from '@/components/Typing';

describe('Typing', () => {
  it('renders as expected', () => {
    const { container } = render(<Typing />);
    expect(container).toMatchSnapshot();
  });
});
