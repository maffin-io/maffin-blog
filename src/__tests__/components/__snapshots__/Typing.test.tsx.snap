// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typing renders as expected 1`] = `
<div>
  <p
    class="overflow-hidden whitespace-nowrap font-mono text-lg leading-7 text-gray-500 animate-typing"
  >
    <span>
      everything
       
    </span>
    <span
      class="text-emerald-500"
    >
      <a
        href="/tags/travel"
      >
        travel
      </a>
    </span>
    <span>
      ,
       
    </span>
    <span
      class="text-sky-500"
    >
      <a
        href="/tags/engineering"
      >
        engineering
      </a>
    </span>
    <span>
       
      and
       
    </span>
    <span
      class="text-orange-500"
    >
      <a
        href="/tags/finance"
      >
        finance
      </a>
    </span>
  </p>
</div>
`;
